packages/react-server/src/ReactFizzServer.js

```diff
@@ -2456,3 +2465,15 @@ export function abort(request: Request, reason: mixed): void {
     fatalError(request, error);
   }
 }
+
+export function getRequestResources(request: Request): Resources {
+  return request.resources;
+}
+
+// If new resources are identified out of band with a task we can still potentially
+// emit them early by pinging them.
+export function pingRequest(request: Request): void {
+  if (request.pingedTasks.length === 0) {
+    scheduleWork(() => performWork(request));
```

Comment:
This has the same problem of overscheduling tasks. One for each preload() potentially.

Reply:
refactored into `enqueueWork` which uses a boolean to track whether work needs to be scheduled